### ISSUE TITLE
Realm.Results do not function correclty unless in debug mode

### ISSUE THREAD

**Issue by ProfessionalAmateur** [realm/realm-js#3990]
### How frequently does the bug occur?

All the time

### Description

Im upgrading to `Realm 10.8.0` I've created a simple test application that creates a Realm schema in the constructor and then inserts and fetches that result in `componentDidMount`.  If I access the properly directly I can see the values, but I cannot iterate over values using `results.every()` or 'results.values()`, etc....  If I put this application into debug mode using Chrome Debugger everything works correctly.  If feels as if there is a missing React-Native dependency for Realm to work properly on an Android device?

**package.json**  

    "react": "17.0.2",
    "react-native": "0.65.1",
    "realm": "^10.8.0"

**App.js**

		import React from 'react';
		import { View, StyleSheet, Text } from 'react-native';
		import Realm from 'realm';

		class Workflows {
			ExternalName;
			InternalName;
			Visible;
			Endpoint;
			Fields;

			static schema = {
				name: 'Workflows',
				primaryKey: 'InternalName',
				properties: {
					ExternalName: 'string',
					InternalName: 'string',
					Visible: 'bool',
					Endpoint: 'string'
				}
			};
		}

		export default class App extends React.Component {
			constructor() {
				super();
				this.realm = new Realm({
					schema: [Workflows],
					schemaVersion: 1,
				});
				console.log("Db: Realm created");

			}

			componentDidMount () {
				// INSERT record //
				let workflows = Object.assign(Object.create(Object.getPrototypeOf(Workflows)), Workflows)
				workflows.InternalName = "Internal_Name";
				workflows.ExternalName = "Readable_Name";
				workflows.Endpoint = "endpoint";
				workflows.Visible = true;
				console.log(JSON.stringify(workflows))

				try {
					this.realm.write(() => {
						this.realm.create(Workflows.schema.name, workflows, "modified");
					})
				}
				catch (error) {
					console.log(error)
				}

				let results;
				let workflow_results;
				results = this.realm.objects(Workflows.schema.name)
				results.map((x) => {
					console.log("InternalName: " + x.InternalName)
					console.log("ExternalName: " + x.ExternalName)
					console.log("Endpoint: " + x.Endpoint)
					console.log("Visible: " + x.Visible)
					console.log("object x:", x) //Object shows as empty
				});
				workflow_results = results.map((x) => ({ title: x.InternalName, ...x }))  //Spread operator doesnt work
				console.log("workflows mapped: " + JSON.stringify(workflow_results))  // Missing all properties from spread operator
				this.setState({ workflowList: workflow_results })
			}

			state = {
				workflowList: []
			}

			render () {
				return (
					<View style={styles.appContainer} >
						<Text>Placeholder</Text>
					</View>
				);
			}
		}

		const styles = StyleSheet.create({
			appContainer: {
				flex: 1,
				alignItems: 'center',
				justifyContent: 'center',
				flexDirection: 'column',
			}
		})

### Stacktrace & log output

```shell
LOG  Running "Test" with {"rootTag":1}
 LOG  Db: Realm created
 LOG  {"schema":{"name":"Workflows","primaryKey":"InternalName","properties":{"ExternalName":"string","InternalName":"string","Visible":"bool","Endpoint":"string"}},"InternalName":"Internal_Name","ExternalName":"Readable_Name","Endpoint":"endpoint","Visible":true}
 LOG  results: {}
 LOG  InternalName: Internal_Name
 LOG  ExternalName: Readable_Name
 LOG  Endpoint: endpoint
 LOG  Visible: true
 LOG  object x: {}
 LOG  workflows mapped: [{"title":"Internal_Name"}]
```

### Can you reproduce the bug?

Yes, always

### Reproduction Steps

1. Create React-Native Project
2. Add Realm
3. Run app (see above), create realm, insert record, retrieve record, attempt to use results

### Version

10.8.0

### What SDK flavour are you using?

Local Database only

### Are you using encryption?

No, not using encryption

### Platform OS and version(s)

Android 7.1.1

### Build environment

Which debugger for React Native: ..
Chrome

**Comment by takameyer** [comment]
@ProfessionalAmateur you should be able to use `.toJSON` on the realm object to view its context in the debugger.  This transforms it into a plain javascript object, which can be used for logging.  Does this help?

**Comment by ProfessionalAmateur** [comment]
@takameyer Not really.  We are upgrading from 2.2.8 to 10.8.0 and all this stuff worked previously.  We cannot perform tasks like iterate over result sets etc..  The example over is just a dummy app I created to show the issue, very simple straight forward example of something I think should work.  The real project uses realm and realm results all over that is breaking with the upgrade.  I wanted to open the issue to see if Im just doing something clearly wrong, or if there is a big in Realm, or if the upgrades introduced functionality changes where I cannot use the product as I expected to anymore.

**Comment by takameyer** [comment]
@ProfessionalAmateur Unfortunately the spread operator is not working since our migration to N-API.  There are [workarounds](https://github.com/realm/realm-js/issues/2844), but unfortunately it may take some time before we get all these parts ironed out.  We do have a [PR for fixing the spread operator](https://github.com/realm/realm-js/pull/3805).
For the other issues I have two more questions.
Do you have Hermes activated?
Are you using react-native-reanimated v2?

**Comment by ProfessionalAmateur** [comment]
@takameyer  - Thanks for the reply.  That is a bummer about the spread operator.  I will need to find more info on the N-API as for this is first time Im hearing about it, but good to know.  
  
Answers to your questions:
>  Do you have Hermes activated?  
No  
  
> Are you using react-native-reanimated v2?  
No, my example above is a plain RN project that only reproduces the issue for the sake of isolation.  But the main project we are upgrading also does not use that package.

Thanks!

**Comment by takameyer** [comment]
What concerns me is that the `object x:` log is outputting an empty object.  That should be showing the contents of the object, which appear to have been set.  We will investigate and see if we can reproduce it.  
I have witnessed in Android this happening, but only when react-native-reanimated v2 is in the project.

**Comment by kneth** [comment]
We have released v11.0.0 which has a complete rewrite of the integration with JSC. I am closing the issue, and ask you to upgrade. If the issue is still observed, please create a new issue.
